apiserver/pkg/rag/rag.go

```diff
@@ -628,3 +628,24 @@ func DeleteRAG(ctx context.Context, kubeClient client.Client, input *generated.D
 	}
 	return kubeClient.DeleteAllOf(ctx, &evav1alpha1.RAG{}, opts...)
 }
+
+func DuplicateRAG(ctx context.Context, kubeClient client.Client, input *generated.DuplicateRAGInput) (*generated.Rag, error) {
+	rag := &evav1alpha1.RAG{}
+	err := kubeClient.Get(ctx, types.NamespacedName{Namespace: input.Namespace, Name: input.Name}, rag)
+	if err != nil {
+		return nil, err
+	}
+	rag.Name = generateKubernetesResourceName("rag", 10)
+	if input.DisplayName != nil {
+		rag.Spec.DisplayName = *input.DisplayName
+	}
```

Comment:
Also need to set up the creator.

https://github.com/kubeagi/arcadia/blob/303c12278739924273c740ca427d51058bf3ad97/apiserver/pkg/rag/rag.go#L286